zerver/migrations/0501_mark_introduce_zulip_view_modals_as_read.py

```diff
@@ -0,0 +1,73 @@
+# Generated by Django 4.2.10 on 2024-03-11 04:43
+
+from django.db import connection, migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import Q
+from django.utils.timezone import now as timezone_now
+from psycopg2.sql import SQL
+
+
+def mark_introduce_zulip_view_modals_as_read(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    with connection.cursor() as cursor:
+        cursor.execute(SQL("SELECT MAX(id) FROM zerver_userprofile WHERE is_bot = False;"))
+        (max_id,) = cursor.fetchone()
+
+    if max_id is None:
+        return
+
+    BATCH_SIZE = 10000
+    lower_id_bound = 0
+    timestamp_value = timezone_now()
+    while lower_id_bound < max_id:
+        upper_id_bound = min(lower_id_bound + BATCH_SIZE, max_id)
+        with connection.cursor() as cursor:
+            query = SQL("""
+                INSERT INTO zerver_onboardingstep (user_id, onboarding_step, timestamp)
+                SELECT id, 'intro_inbox_view_modal', %(timestamp_value)s
+                FROM zerver_userprofile
+                WHERE is_bot = False
+                AND id > %(lower_id_bound)s AND id <= %(upper_id_bound)s
+                UNION ALL
+                SELECT id, 'intro_recent_view_modal', %(timestamp_value)s
+                FROM zerver_userprofile
+                WHERE is_bot = False
```

Comment:
This should probably include `AND is_mirror_dummy = False` -- mirror dummies are usually imported users who haven't used Zulip, and should be eligible for these onboarding items.

Reply:
Done.